loopout.go

```diff
@@ -717,13 +734,75 @@ func (s *loopOutSwap) payInvoice(ctx context.Context, invoice string,
 	return resultChan
 }
 
+func (s *loopOutSwap) payInvoiceWithAssetClient(ctx context.Context,
+	invoice string) (*lndclient.PaymentStatus, error) {
+
+	totalPaymentTimeout := s.executeConfig.totalPaymentTimeout
+
+	sendReq := &routerrpc.SendPaymentRequest{
+		PaymentRequest: invoice,
+		TimeoutSeconds: int32(totalPaymentTimeout.Seconds()),
+		FeeLimitMsat:   1_000_000,
+	}
+
+	paymentStream, err := s.assets.SendPayment(
+		ctx, &tapchannelrpc.SendPaymentRequest{
+			AssetId:        s.AssetId,
+			PeerPubkey:     s.AssetEdgeNode,
+			PaymentRequest: sendReq,
+		})
+	if err != nil {
+		return nil, err
+	}
+
+	// We want to receive the accepted quote message first, so we know how
+	// many assets we're going to pay.
+	for {
+		select {
+		case <-ctx.Done():
+			return nil, ctx.Err()
+		default:
+			msg, err := paymentStream.Recv()
+			if err != nil {
+				return nil, err
+			}
+
+			switch msg.GetResult().(type) {
+			case *tapchannelrpc.SendPaymentResponse_AcceptedSellOrder:
+				quote := msg.GetAcceptedSellOrder()
```

Comment:
don't we have to handle the accepted quote to know how many assets we are going to pay, as the comment suggests?